### PR TITLE
Feature/aip 7419 ark survey results

### PR DESCRIPTION
**Pull request by bennyvdhoogen**

### CODE REVIEW

**Comment by bennyvdhoogen** [comment]
Depending on the migrations in this MR: https://git.data.amsterdam.nl/bmi/aip/-/merge_requests/1520

File: src/schema/ark-survey/__stubs__/geography-data-stub.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	deleted_at: null,
+};
+
+export const createArkSurveyInput = Object.keys(arkSurveyRaw).reduce((input, key) => {
+	input[key] = arkSurveyRaw[key];
+	return input;
+}, new CreateArkSurveyInput());
+
+const updateArkSurvey = new UpdateArkSurveyInput();
+//updateArkSurvey.id = '1f728e79-1b89-4333-a309-ea93bf17667c';
```
Comment (fortunato) [inline src/schema/ark-survey/__stubs__/geography-data-stub.ts:36]: Can this be removed?

File: src/schema/ark-survey/__stubs__/geography-data-stub.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+export const updateArkSurveyInput = Object.keys(arkSurveyRaw).reduce((input, key) => {
+	input[key] = arkSurveyRaw[key];
+	return input;
+}, updateArkSurvey);
+
+const saveArkSurveyInput = new SaveArkSurveyInput();
+//saveArkSurveyInput.id = '1f728e79-1b89-4333-a309-ea93bf17667c';
```
Comment (fortunato) [inline src/schema/ark-survey/__stubs__/geography-data-stub.ts:43]: Can this be removed?

File: src/schema/ark-survey/__stubs__/geography-data-stub.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { ArkSurvey as DomainArkSurvey } from '../types/ark-survey.repository.interface';
+import { CreateArkSurveyInput } from '../dto/create-ark-survey.input';
+import { ArkSurveyFactory } from '../ark-survey.factory';
+import { UpdateArkSurveyInput } from '../dto/update-ark-survey.input';
+import { SaveArkSurveyInput } from '../dto/save-ark-survey.input';
+
+const arkSurveyRaw: Omit<DomainArkSurvey, 'id'> = {
+	surveyId: 'ad18b7c4-b2ef-4e6e-9bbf-c33360584cd7',
+	ArkGeographyStart: {
```
Comment (fortunato) [inline src/schema/ark-survey/__stubs__/geography-data-stub.ts:9]: As discussed on slack. Let's drop the `Ark` prefix and in the future drop the capitalization and stick to camelcase.

File: src/schema/ark-survey/ark-survey.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	@Roles({ roles: ['realm:aip_owner', 'realm:aip_admin', 'realm:aip_survey'], mode: RoleMatchingMode.ANY })
+	public async saveArkSurvey(@Args('saveArkSurvey') input: UpdateArkSurveyInput): Promise<ArkSurvey> {
+		const domainArkSurvey: DomainArkSurvey = await this.commandBus.execute<SaveArkSurveyCommand>(
+			new SaveArkSurveyCommand(input),
+		);
+		return ArkSurveyFactory.createArkSurvey(domainArkSurvey);
+	}
+
+	@Mutation(() => ArkSurvey)
+	@Roles({ roles: ['realm:aip_owner', 'realm:aip_admin', 'realm:aip_survey'], mode: RoleMatchingMode.ANY })
```
Comment (fortunato) [inline src/schema/ark-survey/ark-survey.resolver.ts:54]: Are surveyors allowed to delete a survey?

**Review by fortunato — CHANGES_REQUESTED** [review]
Neat! Few minor thingies

File: src/schema/ark-survey/__stubs__/geography-data-stub.ts (hunk 3 above)
Comment (bennyvdhoogen) [reply src/schema/ark-survey/__stubs__/geography-data-stub.ts:9]: Yes, will do. Also renaming the columns in the db via a migration in the AIP repository.

File: src/schema/ark-survey/ark-survey.resolver.ts (hunk 4 above)
Comment (bennyvdhoogen) [reply src/schema/ark-survey/ark-survey.resolver.ts:58]: Ah, I suppose `realm:aip_survey` means ANY surveyor. Must be just the owner. I'll remove `realm:aip_survey`.

**Review by anaim — APPROVED** [review]

**Pull request merged**